### ISSUE TITLE
[SoundCloud] Search results repeat themselves when scrolling to next page

### ISSUE THREAD

**Issue by bruwalfas** [TeamNewPipe/NewPipe#1632]
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
- [ ] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

I'm using NewPipe v0.13.7, and when I try to search some term in SoundCloud beta mode it shows just one results page, and if I scroll to the bottom to load more, previous results appear. Probably it's because it always asks for the first x results, and the second time it doesn't load the first ones next to the previous ones...
I'm not a Java programmer, but in my opinion I guess this bug can be solved with a simple variable which marks where to start the next search from, instead of looking for the query always from zero.

**Comment by wb9688** [comment]
Since what version does this happen?

**Comment by bruwalfas** [comment]
I started to test it with earlier versions, now trying with different search queries, and I'm starting to see it like another thing.

Sometimes, when searching for a subject, only some results are shown again while other new ones appear—this happens from 0.13.6 to 0.13.7.

What I found interesting is that in version 0.13.5, when looking for more results, in "Video" filter mode, it displays the error "No results", without repeating any result (which is what I find apropiate, but whithout the whole screen error). But if I do the same without filters, results begin to repeat themselves, like in versions 0.13.(6|7).

My idea for the next version is to use the way "Video" mode works in 0.13.5, but instead of showing an error, display in the bottom a little message, like "You've reached the end" or something like that.
I guess SoundCloud API has search results limitations, so it would be nearly impossible to show all the results, like in the web version.

> Since what version does this happen?
